### PR TITLE
NBS #179: use durable client to retry network errors in vhost-server rdma backend

### PR DESCRIPTION
**Pull request by budevg**

### CODE REVIEW

File: cloud/blockstore/vhost-server/backend_rdma.cpp [hunk 1]
```diff
@@ -22,19 +29,114 @@
...
+    TFuture<NProto::T##name##Response> name(                                   \
+        TCallContextPtr callContext,                                           \
+        std::shared_ptr<NProto::T##name##Request> request) override            \
+    {                                                                          \
+        return Storage->name(std::move(callContext), std::move(request));      \
+    }                                                                          \
+// BLOCKSTORE_IMPLEMENT_METHOD
+
+    BLOCKSTORE_IMPLEMENT_METHOD(ReadBlocksLocal)
+    BLOCKSTORE_IMPLEMENT_METHOD(WriteBlocksLocal)
```
Comment (sharpeye) [inline cloud/blockstore/vhost-server/backend_rdma.cpp:101]: [BLOCKSTORE_SERVICE](https://github.com/ydb-platform/nbs/blob/main/cloud/blockstore/libs/service/request.h#L126) разворачивается во что-то такое:
```
BLOCKSTORE_SERVICE
    BLOCKSTORE_STORAGE_SERVICE
        BLOCKSTORE_GRPC_STORAGE_SERVICE
        BLOCKSTORE_LOCAL_SERVICE
            ReadBlocksLocal
            WriteBlocksLocal
    BLOCKSTORE_ENDPOINT_SERVICE
```
т.е. можно в TStorageDataClient сделать:
```

#define BLOCKSTORE_METHOD_NOT_IMPLEMENTED(name, ...)                           \
    TFuture<NProto::T##name##Response> name(                                   \
        TCallContextPtr callContext,                                           \
        std::shared_ptr<NProto::T##name##Request> request) override            \
    {                                                                          \
        Y_UNUSED(callContext);                                                 \
        Y_UNUSED(request);                                                     \
        const auto& type = GetBlockStoreRequestName(EBlockStoreRequest::name); \
        return MakeFuture<NProto::T##name##Response>(TErrorResponse(           \
            E_NOT_IMPLEMENTED,                                                 \
            TStringBuilder() << "Unsupported request " << type.Quote()));      \
    }                                                                          \
// BLOCKSTORE_METHOD_NOT_IMPLEMENTED

#define BLOCKSTORE_IMPLEMENT_METHOD(name, ...)                                 \
    TFuture<NProto::T##name##Response> name(                                   \
        TCallContextPtr callContext,                                           \
        std::shared_ptr<NProto::T##name##Request> request) override            \
    {                                                                          \
        return Storage->name(std::move(callContext), std::move(request));      \
    }                                                                          \
// BLOCKSTORE_IMPLEMENT_METHOD

BLOCKSTORE_GRPC_STORAGE_SERVICE(BLOCKSTORE_METHOD_NOT_IMPLEMENTED)
BLOCKSTORE_ENDPOINT_SERVICE(BLOCKSTORE_METHOD_NOT_IMPLEMENTED)
BLOCKSTORE_LOCAL_SERVICE(BLOCKSTORE_IMPLEMENT_METHOD)
```
и обойтись без TClientBase

**Review by sharpeye — APPROVED** [review]

File: cloud/blockstore/libs/storage/init/disk_agent/actorsystem.cpp [hunk 2]
```diff
@@ -161,7 +161,6 @@ IActorSystemPtr CreateDiskAgentActorSystem(const TDiskAgentActorSystemArgs& daAr
     servicesMask.EnableIcbService = 1;
     servicesMask.EnableLocalService = 0;    // configured manually
     servicesMask.EnableNodeIdentifier = 1;
-    servicesMask.EnableConfigsDispatcher = 1;
```
Comment (drbasic) [inline cloud/blockstore/libs/storage/init/disk_agent/actorsystem.cpp:164]: это  же не относится к делу?

Comment (budevg) [reply cloud/blockstore/libs/storage/init/disk_agent/actorsystem.cpp:164]: Да только первы комит нужно ревюить я потом зделаю rebase

Comment (budevg) [reply cloud/blockstore/libs/storage/init/disk_agent/actorsystem.cpp]: заребейсил

File: cloud/blockstore/vhost-server/backend_rdma.cpp [hunk 3]
```diff
@@ -22,19 +29,102 @@
...
+public:
+    TRetryPolicy(NClient::TClientAppConfigPtr config)
+        : Impl(NClient::CreateRetryPolicy(std::move(config)))
+    {}
+
+    NClient::TRetrySpec ShouldRetry(
+        NClient::TRetryState& state,
+        const NProto::TError& error) override
+    {
+        if (error.GetCode() == E_BS_INVALID_SESSION) {
```
Comment (tpashkin) [inline cloud/blockstore/vhost-server/backend_rdma.cpp:108]: Why do we treat session error as retriable?

Comment (budevg) [reply cloud/blockstore/vhost-server/backend_rdma.cpp:108]: I saw this error coming sometimes when we restart disk agent
In the regular nrd based volume flow path this error is handled by remount of the volume and retry https://github.com/ydb-platform/nbs/blob/65c25012afcc9da5fa0b3decf3d1fd1d188992f6/cloud/blockstore/libs/client/session.cpp#L947

With the vhost-rdma based data path we rely on the rdma partition actor which is running but have no io's going through it to remount the volume after some timeout and then the request will be executed correctly

**Review by tpashkin — APPROVED** [review]

**Review by Sazonov99 — APPROVED** [review]

**Pull request merged**